### PR TITLE
update/0.4.0

### PR DESCRIPTION
**Pull request by decanus**

### CODE REVIEW

File: Sources/BeaconChain/DataStructures/State/BeaconState.swift [hunk 1]
```diff
@@ -31,5 +31,5 @@ struct BeaconState {
 
     var latestEth1Data: Eth1Data
     var eth1DataVotes: [Eth1DataVote]
-    let depositIndex: UInt64
+    var depositIndex: UInt64
```
[the remark this answers is not in the document]
Comment (djrtwo) [reply Sources/BeaconChain/DataStructures/State/BeaconState.swift]: Seems moderately inconsistent throughout

**Comment by djrtwo** [comment]
whatup @gakonst

File: Sources/BeaconChain/BeaconChain.swift [hunk 2]
```diff
@@ -355,13 +354,7 @@ extension BeaconChain {
...
 
         for deposit in genesisValidatorDeposits {
-            processDeposit(
-                state: &state,
-                pubkey: deposit.depositData.depositInput.pubkey,
-                amount: deposit.depositData.amount,
-                proofOfPossession: deposit.depositData.depositInput.proofOfPossession,
-                withdrawalCredentials: deposit.depositData.depositInput.withdrawalCredentials
-            )
+            processDeposit(state: &state, deposit: deposit)
```
[the remark this answers is not in the document]
Comment (decanus) [reply Sources/BeaconChain/BeaconChain.swift]: indeed it was, and i changed it twice in my code

File: Sources/BeaconChain/DataStructures/State/BeaconState.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (decanus) [reply Sources/BeaconChain/DataStructures/State/BeaconState.swift]: `let` can't be mutated

**Pull request merged**